tests/functions.rs

```diff
@@ -412,6 +412,98 @@ test! {
     stderr: "echo Bar\n",
 }
 
+test! {
```

Comment:
These tests are great, but I think we don't need all of them. In general, when using a library, I try to avoid writing tests which test the library itself, instead of testing how it's being used. I just kind of assume the library works, and only test the way it's being used. Since the semver library is handling all the comparisons, I think we can get away with just one test with a matching semver, one with a non-matching semver, and one with an error when the version fails to parse, and one where the req fails to parse.

Reply:
Yeah this is pretty reasonable -- we can simply trust most of what `semver` has here, I'll do a matching `gte` which I think I think is probably the most common

Reply:
OK so for this I left in the gte test!